### ISSUE TITLE
panic: send on closed channel

### ISSUE THREAD

**Issue by goslingwan** [topfreegames/pitaya#132]
## Versions

* Pitaya:1.1.1

## Summary

```
2020-07-10T21:49:30.860+0800	DEBUG	service/handler.go:179	Session read goroutine exit, SessionID=1296, UID=%!d(string=1:0a49046f-affc-496b-add6-23eceab2299e)
2020-07-10T21:49:30.860+0800	DEBUG	agent/agent.go:324	Session handle goroutine exit, SessionID=1296, UID=%!d(string=1:0a49046f-affc-496b-add6-23eceab2299e)
panic: send on closed channel

goroutine 10287 [running]:
github.com/topfreegames/pitaya/agent.(*Agent).heartbeat(0xc000694e40)
	C:/Users/Gosling/go/pkg/mod/github.com/topfreegames/pitaya@v1.1.1/agent/agent.go:368 +0x4b7
created by github.com/topfreegames/pitaya/agent.(*Agent).Handle
	C:/Users/Gosling/go/pkg/mod/github.com/topfreegames/pitaya@v1.1.1/agent/agent.go:328 +0x9e
```

**Comment by leohahn** [comment]
What is the scenario of that happening?

**Comment by felipejfc** [comment]
it seems to be right when the player disconnects

**Comment by chgz** [comment]
when a high-load frondend server, 
close a large number of connections at the same time, there is a probability of the frondend panic

reason is to writing data to chSend
`a.chSend <- pendingWrite{data: hbd}`

my temporary plan is:
`			if a.GetStatus() != constants.StatusClosed {
				a.chSend <- pendingWrite{data: hbd}
			}`

and add recover for heartbeat gorountine.

**Comment by chgz** [comment]
i think， this question was fixed @ PR #148 
@felipejfc  @goslingwan
